### ISSUE TITLE
Public assets included twice in dist package.

### ISSUE THREAD

**Issue by Gacko** [playframework/playframework#5765]
### Play Version (2.5.x / etc)

2.4.6
### API (Scala / Java / Neither / Both)

Scala
### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)

OS X 10.11.3
### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)

```
java version "1.8.0_72"
Java(TM) SE Runtime Environment (build 1.8.0_72-b15)
Java HotSpot(TM) 64-Bit Server VM (build 25.72-b15, mixed mode)
```
### Library Dependencies

```
libraryDependencies ++= Seq(
  "com.typesafe.play" %% "play-slick" % "1.1.1",
  "mysql" % "mysql-connector-java" % "5.1.38"
)
```
### Expected Behavior

I'm having several background images in different sizes included in [public/img/bg](https://github.com/Gacko/Nepenthes/tree/master/public/img/bg).

I expect them to be included only once in the assets JAR and not be included in the application's JAR a second time.
### Actual Behavior

All my images are included in the application's JAR file AND in the assets JAR file which doubles the file size of my distribution ZIP.

```
102M    nepenthes.nepenthes-1.0-assets.jar
102M    nepenthes.nepenthes-1.0-sans-externalized.jar
```

```
236M    nepenthes-1.0.zip
```
### Reproducible Test Case

You can check out my repository (https://github.com/Gacko/Nepenthes) and run dist.

**Comment by Gacko** [comment]
Is there something else I need to do?

**Comment by sebastienbonami** [comment]
I'm in the same situation. My assets are in the **public** folder of the assets.jar file and also in the **META-INF\resources\webjars\\_name_\\_version_** of the sans-externalized.jar file. Is it the expected behavior?

**Comment by salemsaberhagen** [comment]
I'm having the same problem. First I tried naively to exclude any of those jars, but the "externalized" one contains some of your app code also, and the other is the one Play really uses to serve the assets...

Currently I am using this piece of code adapted from [this post](https://github.com/sbt/sbt-assembly/issues/94#issuecomment-24806234) and it seems to work:

```
mappings in (Compile, packageBin) ~= { (ms: Seq[(File, String)]) =>
    ms filterNot { case (file, dest) =>
        dest.contains("resources/webjars")
    }
}
```

**Comment by anfuerer** [comment]
+1 for fixing this issue soon. If you use a PaaS service like Heroku that has a 300MB limit of application deployment artifact this can be quite blocking for deployments.

**Comment by schmitch** [comment]
well I can confirm that however not every assets are included.
it looks like only the "raw" assets (not the one's gone trough the **production** pipeline are there twice).

**Comment by mirikle** [comment]
Same issue happened here, it gets worse for a project that has over 50MB+ of files in the /public directory.

**Comment by Gacko** [comment]
Can someone confirm this issue on latest releases?

**Comment by dzagorovsky** [comment]
The same behaviour with sbt 0.13.15

**Comment by mkurz** [comment]
@Gacko why did you close this issue, can you confirm it is fixed? Since multiple people confirm this issue, we should make sure it was fixed before closing.

**Comment by mkurz** [comment]
The origin of this problem is the same as in #5242:
The solution is to set
```sbt
Assets / WebKeys.exportedMappings in := Seq()
TestAssets / WebKeys.exportedMappings in := Seq()
```
We will make this default in an upcoming Play release.

**Comment by mkurz** [comment]
So I did some testing with the fix (exportedMappings = Nil) locally and the nice side effect of it is that if you have many files and/or big files inside `assets/` and/or `public/` a reload in dev mode will now be 50% faster :+1: See my pseudo benchmark here: https://github.com/playframework/playframework/issues/7799#issuecomment-1999742140
(Like mentioned in the referenced ticket, the 50% is/was kind of predictable because with a fix for this PR and #5242 the assets pipeline does not run twice anymore and also the assets will not be copied twice anymore in the target folder...)

**Comment by mkurz** [comment]
Since the fix is a new setting in sbt-web, you can make use of it right now already:
In `build.sbt` you need to set:
```sbt
Assets / WebKeys.webJarsAddMappingsToPackageBin     := false
TestAssets / WebKeys.webJarsAddMappingsToPackageBin := false
```
Also, you need to upgrade to sbt-web 1.5.7, so in `project/plugins.sbt`:
```sbt
// Upcoming Play 2.9.4 and 3.0.4 will ship with this version out of the box, so this can then be removed
addSbtPlugin("com.github.sbt" % "sbt-web" % "1.5.7")
```

Please test your packaged application however. We will not activate the fix out of the box before Play 3.1.0.

**Comment by mkurz** [comment]
> Since the fix is a new setting in sbt-web, you can make use of it right now already: In `build.sbt` you need to set:
> 
> ```scala
> Assets / WebKeys.webJarsAddMappingsToPackageBin     := false
> TestAssets / WebKeys.webJarsAddMappingsToPackageBin := false
> ```

With sbt-web 1.5.8 this setting was renamed to `addExportedMappingsToPackageBinMappings` because that naming makes much more sense.
